src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,211 @@
+package duke;
+import java.io.*;
+import java.util.Scanner;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.InvalidCommandException;
+import java.util.ArrayList;
+
+import duke.utilityfunctions.Filereader;
+public class Duke {
+    public static void generateTaskStatus(String taskIcon, String statusIcon, String description) {
+        System.out.println("\t[" + taskIcon + "]" + "[" + statusIcon + "] " + description);
+    }
+    public static void drawLine() {
+        System.out.println("------------------------------------");
+    }
+    public static void addedMsg() {
+        System.out.println("Got it. I've added this task:");
+    }
+
+    public static void taskCountReminder(int noOfTasks) {
+        System.out.printf("Now you have %d tasks in the list.\n",noOfTasks);
+    }
+
+    public static boolean isValidCommand(String s){
+        String[] validCommandArray = {"bye", "list","unmark","mark","todo","deadline","event", "delete"};
+        for(int i = 0; i < validCommandArray.length; i++) {
+            if(s.equals(validCommandArray[i])){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static void loadTasksFromTextFile(String filepath, ArrayList<Task> tasks) throws FileNotFoundException {
+        File f = new File(filepath);
+        Scanner s = new Scanner(f);
+        Filereader textReader = new Filereader();
+        while(s.hasNext()) {
+            tasks.add(textReader.generateTask(s.nextLine()));
+        }
+        s.close();
+    }
+
+    public static void updateFile(String filepath, ArrayList<Task> taskArray) throws IOException {
+        FileWriter fw = new FileWriter(filepath);
+        for(int i = 0; i < taskArray.size(); i++){
+            if(i != taskArray.size() - 1){
+                fw.write("[" + taskArray.get(i).getTaskIcon() + "]" + "[" + taskArray.get(i).getStatusIcon() + "] " + taskArray.get(i).getRawDescription() + System.lineSeparator());
+            } else {
+                fw.write("[" + taskArray.get(i).getTaskIcon() + "]" + "[" + taskArray.get(i).getStatusIcon() + "] " + taskArray.get(i).getRawDescription());
+            }
+        }
+        fw.close();
+    }
+
+    public static void main(String[] args) {
```

Comment:
This function is too long (>30 lines). Please refactor the code and create multiple smaller functions to be called in this main function.